server/server.js

```diff
@@ -1,56 +1,88 @@
+ * bodies in a middleware before you handle it.
+ */
+
+app.use(function (req, res, next) {
     //add a header message to solve cors security policy access deny problem
     res.header("Access-Control-Allow-Origin", "http://localhost:3000"); // update to match the domain you will make the request from
     res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    res.header("Access-Control-Allow-Methods", "POST, GET, DELETE, OPTIONS");
     next();
-  });
+});
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
 
-app.get('/', async (req, res, next) =>{
-    console.log(req.body);
-    db.query('select * from master where contained_by = 0', [],function(result,fields){
+
+//home page's data
+app.get('/', async (req, res, next) => {
+    db.query('select * from master where contained_by = 0', [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end( JSON.stringify(result));
+        res.end(JSON.stringify(result));
     });
 });
 
-app.get('/:id',function(req,res, next){
+//get specific item's data
+app.get('/:id', function (req, res, next) {
     var id = req.params.id;
-    db.query(`select * from master where contained_by = ${id}`, [],function(result,fields){
+    db.query(`select * from master where contained_by = ${id}`, [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end( JSON.stringify(result));
+        res.end(JSON.stringify(result));
     });
 });
 
-app.post('/addItem', function (req, res) {
-    db.query('select * from master', [],function(result,fields){
-        var toAdd = req.body;
+//get current container info
+app.get('/containerInfo/:id', function (req, res, next) {
+    var id = req.params.id;
+    db.query(`select * from master where item_id = ${id}`, [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end();
+        res.end(JSON.stringify(result));
     });
- });
+});
 
- app.post('/container', function (req, res) {
-    db.query('select * from master', [],function(result,fields){
-        var toAdd = req.body;
-        console.log('查询结果：');
+//add new item
+app.post('/addItem', function (req, res) {
+    console.log("request body:")
+    console.table(req.body);
+    var name = req.body.item_name;
+    var adInfo = req.body.additional_information;
+    console.log("is container? "+ req.body.is_container);
+    var isC = req.body.is_container==='true' ? 1 : 0;
+    var cB = parseInt(req.body.contained_by); 
+    db.query("INSERT INTO master (item_name, contained_by, is_container, additional_json) VALUES (?,?,?,?);", [name, cB, isC, adInfo], function (result, fields) {
+        console.log('查询Added Item：');
         console.log(result);
 
+        res.end(JSON.stringify(result));
+    });
+});
+
+// delete item
+app.delete('/deleteItem/:id', function (req, res) {
```

Comment:
Also, you can handle delete request by `/item/:id` url as well.
```suggestion
app.delete('/item/:id', function (req, res) {
```